File: tests/ground_truth/code/test_service.kt
// Kotlin Test File
package com.example.test

data class User(val name: String, val email: String, val age: Int)

class UserService {
    private val users = mutableListOf<User>()
    
    fun addUser(user: User) = users.add(user)
    
    fun findByName(name: String): User? = users.find { it.name == name }
}

fun main() {
    val service = UserService()
    service.addUser(User("Max", "max@example.com", 30))
    println("User Service gestartet")
}

File: tests/ground_truth/code/test_service.php
<?php
// PHP Test File 1: Classes
// Expected: Extract class definitions

namespace App\Services;

class UserService {
    private \ = [];
    
    public function addUser(string \, string \): void {
        \->users[] = [
            'name' => \,
            'email' => \,
            'created_at' => date('Y-m-d H:i:s')
        ];
    }
    
    public function findByEmail(string \): ?array {
        foreach (\->users as \) {
            if (\['email'] === \) {
                return \;
            }
        }
        return null;
    }
    
    public function count(): int {
        return count(\->users);
    }
}

\ = new UserService();
\->addUser('Max', 'max@example.com');
echo "Users: " . \->count();
